### PR TITLE
Use oneapi::dpl::sort_by_key if possible

### LINKED ISSUE #926: OneDPL added sort_by_key
It was recently added in oneapi-src/oneDPL#859. Should probably switch to using that and avoid zipping stuff together as we do right now: 

https://github.com/arborx/ArborX/blob/d05083f6f8d908351ddf7fe5a66884458f6c781f/src/kokkos_ext/ArborX_DetailsKokkosExtSort.hpp#L173C1-L179

The question is whether its performance is any better or worse than what we have right now.

### PR DESCRIPTION
**Pull request by masterleinad**
Fixes #926. I didn't see much of a difference in performance but that also wasn't really expected at this point.

### CODE REVIEW

File: src/kokkos_ext/ArborX_DetailsKokkosExtSort.hpp [hunk 1]
```diff
@@ -170,13 +170,18 @@ void sortByKey(Kokkos::Experimental::SYCL const &space, Keys &keys,
   if (n == 0)
     return;
 
-  auto zipped_begin =
-      oneapi::dpl::make_zip_iterator(keys.data(), values.data());
   oneapi::dpl::execution::device_policy policy(
       *space.impl_internal_space_instance()->m_queue);
+#if ONEDPL_VERSION_MAJOR > 2022 ||                                             \
+    (ONEDPL_VERSION_MAJOR == 2022 && ONEDPL_VESION_MINOR >= 2)
```
Comment (aprokop) [inline src/kokkos_ext/ArborX_DetailsKokkosExtSort.hpp:176]: Should we just require this? What's the minimum version for Kokkos? Do we care?

Comment (masterleinad) [reply src/kokkos_ext/ArborX_DetailsKokkosExtSort.hpp:176]: `Kokkos` doesn't have a minimum version; as of now, we are only using `dpl::sort` anyway.
`oneDPL` is not necessarily recent enough on the testbeds so I wouldn't want to require 2022.2 at this point.

Comment (dalg24) [inline src/kokkos_ext/ArborX_DetailsKokkosExtSort.hpp:176]:
```suggestion
    (ONEDPL_VERSION_MAJOR == 2022 && ONEDPL_VERSION_MINOR >= 2)
```

**Comment by aprokop** [comment]
@masterleinad please confirm that you verified that the code executes the sort by key with the new onedpl. In other words, that the macro check is correct.

**Comment by masterleinad** [comment]
> @masterleinad please confirm that you verified that the code executes the sort by key with the new onedpl. In other words, that the macro check is correct.

I tested without the macro guards before on the testbeds (with newer than default compiler versions, i.e., an `oneDPL` commit that includes `sort_by_key` based on `2022.1.1`), and that worked fine.

**Comment by masterleinad** [comment]
I also tested manually with the 2022.2.0 oneDPL release which works fine. However, we get a bunch more warnings and it's not clear to me if it's beneficial to update the CI (here or in a follow-up pull request).

**Review by aprokop — APPROVED** [review]
LGTM

**Pull request merged**